v1.8.x: kube-proxy panics on cleanup because of nil pointer

**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

Ran `kube-proxy --cleanup` and it crashed after cleaning up `iptables`:

```
# ./kube-proxy --v=10 --cleanup
W1020 16:27:12.840883    2064 server.go:191] WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.
I1020 16:27:12.861382    2064 iptables.go:564] couldn't get iptables-restore version; assuming it doesn't support --wait
I1020 16:27:12.863689    2064 iptables.go:396] running iptables -C [OUTPUT -t nat -m comment --comment handle ClusterIPs; NOTE: this must be before the NodePort rules -j KUBE-PORTALS-HOST]
I1020 16:27:12.864612    2064 iptables.go:396] running iptables -C [PREROUTING -t nat -m comment --comment handle ClusterIPs; NOTE: this must be before the NodePort rules -j KUBE-PORTALS-CONTAINER]
I1020 16:27:12.865435    2064 iptables.go:396] running iptables -C [OUTPUT -t nat -m addrtype --dst-type LOCAL -m comment --comment handle service NodePorts; NOTE: this must be the last rule in the chain -j KUBE-NODEPORT-HOST]
I1020 16:27:12.866281    2064 iptables.go:396] running iptables -C [PREROUTING -t nat -m addrtype --dst-type LOCAL -m comment --comment handle service NodePorts; NOTE: this must be the last rule in the chain -j KUBE-NODEPORT-CONTAINER]
I1020 16:27:12.867102    2064 iptables.go:396] running iptables -C [INPUT -t filter -m comment --comment Ensure that non-local NodePort traffic can flow -j KUBE-NODEPORT-NON-LOCAL]
I1020 16:27:12.867862    2064 iptables.go:396] running iptables -F [KUBE-PORTALS-CONTAINER -t nat]
I1020 16:27:12.868564    2064 iptables.go:396] running iptables -F [KUBE-PORTALS-HOST -t nat]
I1020 16:27:12.869247    2064 iptables.go:396] running iptables -F [KUBE-NODEPORT-HOST -t nat]
I1020 16:27:12.869921    2064 iptables.go:396] running iptables -F [KUBE-NODEPORT-CONTAINER -t nat]
I1020 16:27:12.870622    2064 iptables.go:396] running iptables -F [KUBE-NODEPORT-NON-LOCAL -t filter]
I1020 16:27:12.871304    2064 iptables.go:396] running iptables -C [INPUT -t filter -m comment --comment kubernetes service portals -j KUBE-SERVICES]
I1020 16:27:12.872129    2064 iptables.go:396] running iptables -C [OUTPUT -t filter -m comment --comment kubernetes service portals -j KUBE-SERVICES]
I1020 16:27:12.872884    2064 iptables.go:396] running iptables -C [OUTPUT -t nat -m comment --comment kubernetes service portals -j KUBE-SERVICES]
I1020 16:27:12.873639    2064 iptables.go:396] running iptables -C [PREROUTING -t nat -m comment --comment kubernetes service portals -j KUBE-SERVICES]
I1020 16:27:12.874398    2064 iptables.go:396] running iptables -C [POSTROUTING -t nat -m comment --comment kubernetes postrouting rules -j KUBE-POSTROUTING]
I1020 16:27:12.875133    2064 iptables.go:314] running iptables-save [-t nat]
I1020 16:27:12.875860    2064 iptables.go:373] running iptables-restore [-T nat --noflush --counters]
I1020 16:27:12.876637    2064 iptables.go:373] running iptables-restore [-T filter --noflush --counters]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1287ad8]

goroutine 1 [running]:
k8s.io/kubernetes/pkg/proxy/ipvs.deleteDummyDevice(0x0, 0x0, 0x16076ac, 0xa, 0x15ff1dc, 0x3)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/proxy/ipvs/proxier.go:1496 +0xa8
k8s.io/kubernetes/pkg/proxy/ipvs.CleanupLeftovers(0x0, 0x0, 0x1f763a0, 0xc420342a40, 0x1f76aa0, 0xc4203e4ab0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/proxy/ipvs/proxier.go:720 +0x75
k8s.io/kubernetes/cmd/kube-proxy/app.(*ProxyServer).Run(0xc4200b63c0, 0xc42006d601, 0xc4204c2b80)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:428 +0x10aa
k8s.io/kubernetes/cmd/kube-proxy/app.(*Options).Run(0xc42006d6b0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:229 +0xa0
k8s.io/kubernetes/cmd/kube-proxy/app.NewProxyCommand.func1(0xc4204be6c0, 0xc4201cce60, 0x0, 0x2)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:344 +0x80
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc4204be6c0, 0xc42000e130, 0x2, 0x2, 0xc4204be6c0, 0xc42000e130)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x22b
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4204be6c0, 0xc420645f60, 0xc420645f58, 0xc42010b380)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x339
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4204be6c0, 0x168ecb0, 0xc4200001a0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.main()
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/proxy.go:44 +0x96
```

**What you expected to happen**:

I expected `kube-proxy --cleanup` not to crash.

**How to reproduce it (as minimally and precisely as possible)**:

This can be reproduced this by running the abovementioned command or the shorter

```
# kube-proxy --cleanup
```

**Anything else we need to know?**:

—

**Environment**:
- Kubernetes version (use `kubectl version`): v1.8.1 (all components from the `v1.8.1` tag)
- Cloud provider or hardware configuration: GCE
- OS (e.g. from /etc/os-release): Ubuntu 16.04 LTS
- Kernel (e.g. `uname -a`): `Linux worker-0 4.10.0-37-generic #41~16.04.1-Ubuntu SMP Fri Oct 6 22:42:59 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`
- Install tools: —
- Others: —

--- a/cmd/kube-proxy/app/server_others.go
+++ b/cmd/kube-proxy/app/server_others.go
@@ -82,7 +82,12 @@ func NewProxyServer(config *componentconfig.KubeProxyConfiguration, cleanupAndEx
 
 	// We omit creation of pretty much everything if we run in cleanup mode
 	if cleanupAndExit {
-		return &ProxyServer{IptInterface: iptInterface, IpvsInterface: ipvsInterface, CleanupAndExit: cleanupAndExit}, nil
+		return &ProxyServer{
+			execer:         execer,
+			IptInterface:   iptInterface,
+			IpvsInterface:  ipvsInterface,
+			CleanupAndExit: cleanupAndExit,
+		}, nil
 	}
 
 	client, eventClient, err := createClients(config.ClientConnection, master)

